api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -515,6 +548,21 @@ public <T> Literal<T> to(Type type) {
             return (Literal<T>) new TimestampLiteral(timestampMicros);
           }
 
+        case TIMESTAMP_NANO:
+          if (((Types.TimestampNanoType) type).shouldAdjustToUTC()) {
+            long timestampNanos =
+                ChronoUnit.NANOS.between(
+                    EPOCH, OffsetDateTime.parse(value(), DateTimeFormatter.ISO_DATE_TIME));
+            return (Literal<T>) new TimestampNanoLiteral(timestampNanos);
```

Comment:
Can this be a method in `DateTimeUtil`? I think that's a good place to put it.

Reply:
Done.